source/common/json/config_schemas.cc

```diff
@@ -1016,6 +1016,40 @@ const std::string Json::Schema::CLUSTER_MANAGER_SCHEMA(R"EOF(
   }
   )EOF");
 
+const std::string Json::Schema::LDS_SCHEMA(R"EOF(
+  {
+    "$schema": "http://json-schema.org/schema#",
+    "type" : "object",
+    "properties" : {
+      "listeners" : {
+        "type" : "array",
+        "items" : {"type" : "object"}
```

Comment:
There is no great way right now to share schemas (reference another schema in another variable inside this schema). What we have here is probably the best we can do. In practice, it provides OK coverage since we do verify the returned items are objects, so asking for name will fail cleanly, then yes we do the full schema check inside addOrUpdate.